pkg/network/common/egressip.go

```diff
@@ -249,6 +252,17 @@ func (eit *EgressIPTracker) handleAddOrUpdateHostSubnet(obj, _ interface{}, even
 			return
 		}
 	}
+	if eit.recorder != nil && len(hs.EgressIPs) > 0 && eit.CloudEgressIP {
+		capacity, err := eit.getCapacity(hs)
+		if err != nil {
+			klog.Errorf("Ignoring invalid HostSubnet %s: %v", HostSubnetToString(hs), err)
+		}
+		if len(hs.EgressIPs) >= capacity {
```

Comment:
it's awkward that this test runs even if `getCapacity` returns an error... would be safer to have an `else` here